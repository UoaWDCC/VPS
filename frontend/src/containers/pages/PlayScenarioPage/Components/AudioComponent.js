import React from "react";
import ReactAudioPlayer from "react-audio-player";

function AudioComponent({ key, id, onClick, component }) {
  console.log(component);
  return <ReactAudioPlayer src="my_audio_file.ogg" autoPlay />;
}

export default AudioComponent;
