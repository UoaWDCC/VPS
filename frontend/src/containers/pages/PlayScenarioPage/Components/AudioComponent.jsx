import ReactAudioPlayer from "react-audio-player";

/**
 * This component plays audio from a scene
 * @component
 */
function AudioComponent({ id, component }) {
  // (window.location === window.parent.location) is true if it's not inside of an iframe
  return (
    window.location === window.parent.location && (
      <ReactAudioPlayer
        muted={!!(window.location !== window.parent.location)}
        id={id?.toString()}
        src={`${component.url}`}
        autoPlay
        loop={!!component.loop}
      />
    )
  );
}

export default AudioComponent;
