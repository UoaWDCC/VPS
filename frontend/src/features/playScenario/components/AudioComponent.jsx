import ReactAudioPlayer from "react-audio-player";

/**
 * This component plays audio from a scene
 * @component
 */
function AudioComponent({ id, component }) {
  return (
    <ReactAudioPlayer
      id={id}
      src={component.url}
      autoPlay
      loop={!!component.loop}
    />
  );
}

export default AudioComponent;
