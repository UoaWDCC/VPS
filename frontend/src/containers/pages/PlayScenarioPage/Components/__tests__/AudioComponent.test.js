import React from "react";
import { render } from "@testing-library/react";
import AudioComponent from "../AudioComponent";

test("Add firebase audio component snapshot test", () => {
  const { baseElement } = render(<AudioComponent component={{}} />);

  expect(baseElement).toMatchSnapshot();
});
