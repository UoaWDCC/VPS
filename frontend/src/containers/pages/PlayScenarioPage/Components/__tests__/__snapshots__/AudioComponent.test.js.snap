// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add firebase audio component snapshot test 1`] = `
<body>
  <div>
    <audio
      autoplay=""
      class="react-audio-player "
      id=""
      preload="metadata"
      src="undefined"
      title="undefined"
    >
      <p>
        Your browser does not support the 
        <code>
          audio
        </code>
         element.
      </p>
    </audio>
  </div>
</body>
`;
